examples/tutorials/ProteinLM_Tutorial0.ipynb

```diff
@@ -0,0 +1,2855 @@
+{
```

Comment:
<p>I think we should give some more detail about what a language model is. We should talk about transformers and attention and encoder-decoder here.</p><p> </p><p>Getting probability distribution over tokens is the final output of these models. Its the objective we should also tell about how we reached that objective.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchem/deepchem/pull/4054/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Reply:
<p>I thought about that too, but I thought to keep this as a tutorial 0 (intuition only) for a couple of reasons:</p><p> </p><ol><li>The tutorial would go from a 5-7 minute read to something that spans 15-20 min.</li><li>Without understanding a language model with the details abstracted, a first time reader might get caught up in the whole things</li><li>Language model is an abstraction that sits one layer above the architecture MAMBA is a CausalLM but is a SSM and is competitive w/  transformers.</li></ol><p> </p><p>I agree with you that  a tutorial on architecture would be hugely helpful for the community, but I think keeping it separate might be the better call here. Lmk what you think</p>